jbmc/src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -603,6 +610,43 @@ void java_bytecode_parsert::get_class_refs()
   }
 }
 
+/// For each of the given annotations, get a reference to its class and
+/// recursively get class references of the values it stores.
+void java_bytecode_parsert::get_annotation_refs(
+  const java_bytecode_parse_treet::annotationst &annotations)
+{
+  for(const auto &annotation : annotations)
+  {
+    get_class_refs_rec(annotation.type);
+    for(const auto &element_value_pair : annotation.element_value_pairs)
+    {
+      const exprt &value = element_value_pair.value;
+      get_annotation_refs_rec(value);
+    }
+  }
+}
+
+/// See \ref java_bytecode_parsert::get_annotation_refs.
+/// For the different cases of `exprt`, see \ref
+/// java_bytecode_parsert::get_relement_value.
+void java_bytecode_parsert::get_annotation_refs_rec(const exprt &value)
+{
+  if(const auto &symbol_expr = expr_try_dynamic_cast<symbol_exprt>(value))
+  {
+    const irep_idt &value_id = symbol_expr->get_identifier();
+    const typet value_type = java_type_from_string(id2string(value_id));
+    get_class_refs_rec(value_type);
+  }
+  if(const auto &array_expr = expr_try_dynamic_cast<array_exprt>(value))
```

Comment:
An `else` here would shortcut the check.

Reply:
But that could lead to unexpected behaviour if the `todo`s in `get_relement_value` are addressed but the ones in this function are not? Since an enum or annotation (in the nested annotation case) could be passed as an argument to this function, so just because the argument is not a `symbol_exprt`, we can't just assume that it's an `array_exprt`.

Reply:
Yes, I meant an `else if`, not just an `else`.